### PR TITLE
Fix the export of static types

### PR DESCRIPTION
**Pull request by turbolent**
Static types are currently only exported as a string. However, only the qualified identifier is used (e.g. any type `S` in `C` has qualified identifier `C.S`), which is not unique. 
Use the full type ID instead (e.g. type `S` in `C`, deployed to account 0x1, has ID `A.0x1.C.S`).

Also, add more tests (export of static types which are not primitive)

### CODE REVIEW

File: encoding/json/encoding_test.go [hunk 1]
```diff
@@ -677,15 +677,15 @@ func TestEncodeResource(t *testing.T) {
 		t.Parallel()
 
 		script := `
-			access(all) resource Foo {
-				access(all) let bar: Int
+			pub resource Foo {
```
Comment (janezpodhostnik) [inline encoding/json/encoding_test.go:680]: I thought access(all) and pub were equivalent. Why the change? just for clarity?

**Review by janezpodhostnik — APPROVED** [review]
I think I understand the change and why, but I'm missing a decoding of static types change... or did I misunderstand?

File: encoding/json/encoding_test.go (hunk 1 above)
Comment (turbolent) [reply encoding/json/encoding_test.go:680]: Yeah, all tests use `pub` and I just replaced it for consistency reasons.

**Review by psiemens — APPROVED** [review]

**Comment by turbolent** [comment]
@janezpodhostnik this PR only fixes the export of the static types in links, types, and capabilities, because the import of those values is not supported yet
